Back-To-School/Assets/BackToSchool/Scripts/Inputs/BaseInput.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Assets.BackToSchool.Scripts.Interfaces.Input;
+using UnityEngine;
+
+
+namespace Assets.BackToSchool.Scripts.Inputs
+{
+    public class BaseInput : IInputProvider
```

Comment:
That may be splitted to 2 interfaces. Current - rename to sth like `IRawInputProvider` because it parses raw input data.
And other - `IInputProvider` - interface which should contain specific thinks for generic Input provider - `void TogglePause(bool isPaused);` method

Reply:
So such way you can add to Base input default implementation of pausing ability.